Sources/Basics/FileSystem+Extensions.swift

```diff
@@ -69,52 +68,126 @@ extension FileSystem {
     }
 }
 
-// MARK: - config
+// MARK: - configuration
 
 extension FileSystem {
-    private var idiomaticUserConfigDirectory: AbsolutePath? {
-        return FileManager.default.urls(for: .libraryDirectory, in: .userDomainMask).first.flatMap { AbsolutePath($0.path) }
-    }
-
     /// SwiftPM config directory under user's config directory (if exists)
-    public var swiftPMConfigDirectory: AbsolutePath {
-        if let path = self.idiomaticUserConfigDirectory {
-            return path.appending(component: "org.swift.swiftpm")
+    public var swiftPMConfigurationDirectory: AbsolutePath {
+        if let path = self.idiomaticSwiftPMDirectory {
+            return path.appending(component: "configuration")
         } else {
-            return self.dotSwiftPMConfigDirectory
+            return self.dotSwiftPMConfigurationDirectory
         }
     }
 
-    fileprivate var dotSwiftPMConfigDirectory: AbsolutePath {
-        return self.dotSwiftPM.appending(component: "config")
+    fileprivate var dotSwiftPMConfigurationDirectory: AbsolutePath {
+        return self.dotSwiftPM.appending(component: "configuration")
+    }
+}
+
+extension FileSystem {
+    public func getOrCreateSwiftPMConfigurationDirectory() throws -> AbsolutePath {
+        let idiomaticConfigurationDirectory = self.swiftPMConfigurationDirectory
+
+        // temporary 5.6, remove on next version: transition from previous configuration location
+        do {
+            if !self.exists(idiomaticConfigurationDirectory) {
+                try self.createDirectory(idiomaticConfigurationDirectory, recursive: true)
+            }
+
+            if idiomaticConfigurationDirectory != self.dotSwiftPMConfigurationDirectory {
+                // move content from old directory to new one but create symlink for the files for backwards compatibility (eg older xcode)
+                let oldConfigDirectory = idiomaticConfigurationDirectory.parentDirectory
+                if self.exists(oldConfigDirectory, followSymlink: false) && self.isDirectory(oldConfigDirectory) {
+                    let content = try self.getDirectoryContents(oldConfigDirectory)
+                    for item in content {
+                        if self.isFile(oldConfigDirectory.appending(component: item)) && !self.isSymlink(oldConfigDirectory.appending(component: item)) {
+                            try self.move(from: oldConfigDirectory.appending(component: item), to: idiomaticConfigurationDirectory.appending(component: item))
+                            try self.createSymbolicLink(oldConfigDirectory.appending(component: item), pointingAt: idiomaticConfigurationDirectory.appending(component: item), relative: false)
+                        }
+                    }
+                }
+                // re-point old symlink
+                let oldConfigSymlink = self.dotSwiftPM.appending(component: "config")
```

Comment:
How come we don't need to copy/move files from `~/.swiftpm/config/` to `idiomaticConfigurationDirectory`? Should this be swapped with the logic in `else`-block?